src/compose/ComposeBox.js

```diff
@@ -86,15 +87,83 @@ function randomInt(min, max) {
   return Math.floor(Math.random() * (max - min + 1)) + min;
 }
 
-const updateTextInput = (textInput, text) => {
-  if (textInput === null) {
-    // Depending on the lifecycle events this function is called from,
-    // this might not be set yet.
-    return;
-  }
+/**
+ * State management for an Input component that doesn't get passed `value`
+ *   or `selection`, i.e., an uncontrolled Input.
+ *
+ * For background on the concept, see
+ *   https://reactjs.org/docs/uncontrolled-components.html
+ * and our #2738 for the performance concerns that led us to use this in the
+ * compose box.
+ *
+ * Takes a ref that exposes the underlying TextInput's `setNativeProps`, and
+ * returns:
+ *
+ * - The value and selection state
+ * - Functions to set the value and selection
+ * - Private functions that should be passed directly to the Input as props:
+ *     _handleValueChange as onChangeText
+ *     _handleSelectionChange as onSelectionChange
+ */
+const useUncontrolledInput = (args: {|
+  ref: {| current: $FlowFixMe | null |},
```

Comment:
How about we move the `useRef` call inside `useUncontrolledInput`? It feels like it's naturally part of what this hook is taking care of.

Reply:
I waffled a bit, but would be fine doing it that way. There are other legitimate uses for the ref that one might have in addition to the uses relevant to the hook, which is why I thought the hook might be overstepping a bit by creating/returning the ref, rather than just consuming it. But I don't feel strongly.